sdk/physical/cache.go

```diff
@@ -4,6 +4,8 @@ import (
 	"context"
 	"sync/atomic"
 
+	metrics "github.com/armon/go-metrics"
```

Comment:
If possible, we would like new metrics to go through core.metricSink instead of continuing to use the global metrics object.  It can be passed in as an argument during construction.  (It looks like that is just a couple places in the code.)

Reply:
Should this be accepted into `NewCache` and `NewTransactionCache` as the `MetricSink` interface or as the `metricsutil.ClusterMetricSink` concrete type. I would assume the interface but want to make sure.

Reply:
Hmm.. I believe ClusterMetricSink is not (yet?) a complete implementation of MetricSink, unfortunately.

Using the concrete type might cause problems because of the circular dependency between packages (metricsutil is in the "vault" package, not "sdk".)  If we can't do this right now, I'm happy to approve the PR without it, or with taking a MetricSink as an argument and working on filling in ClusterMetricSink later.

Reply:
You're right. The `Sink` field in the `ClusterMetricSink` is public though. If acceptable we can pass that in and include a comment saying that once the `ClusterMetricSink completes the interface it should be used instead. It should allow us to use the MetricSink interface and not use the global metrics object.

What do you think?

Reply:
Sure, that seems reasonable.

Reply:
Excellent. I'll update the PR within 24 hours.